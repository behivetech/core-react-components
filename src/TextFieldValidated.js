// Vendor Libs
import React, {Component} from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import {omit} from 'lodash';

// Components
import withValidation from './withValidation';

class TextFieldValidated extends Component {
    render() {
        const {className, component} = this.props;
        const TextField = component;

        return (
            <TextField 
                {...omit(this.props, ['formState'])} 
                className={classnames('text-field-validated', className)} 
            />
        );
    }
}

TextFieldValidated.propTypes = {
    className: PropTypes.string,
    component: PropTypes.func,
};

export default withValidation(TextFieldValidated);