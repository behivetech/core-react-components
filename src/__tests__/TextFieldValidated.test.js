import TextFieldValidated from '../TextFieldValidated';

jest.mock('../withValidation');

describe('TextFieldValidated', () => {
    const mockComponent = (props) => <div />;
    const getProps = () => {
        return {
            className: 'mock-class',
            label: 'mock-label',
            name: 'mockName',
            validate: [jest.fn()],
            component: mockComponent,
        };
    };
    
    it('should render with the correct props', () => {
        const wrapper = shallow(<TextFieldValidated {...getProps()} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.dive()).toMatchSnapshot();
    });
});