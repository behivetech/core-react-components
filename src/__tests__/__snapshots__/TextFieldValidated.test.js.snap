// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextFieldValidated should render with the correct props 1`] = `
<TextFieldValidated
  className="mock-class"
  component={[Function]}
  label="mock-label"
  name="mockName"
  validate={
    Array [
      [MockFunction],
    ]
  }
/>
`;

exports[`TextFieldValidated should render with the correct props 2`] = `
<mockComponent
  className="text-field-validated mock-class"
  component={[Function]}
  label="mock-label"
  name="mockName"
  validate={
    Array [
      [MockFunction],
    ]
  }
/>
`;
